File: 219. Contains Duplicate II.cpp
#include"LeetCodeLibs.h"

class Solution {
public:
    bool containsNearbyDuplicate(vector<int>& nums, int k) {
        unordered_map<int, int> seen;
        int n = nums.size();

        for(int i = 0; i < n; ++i){
            if(seen.count(nums[i]) && i - seen[nums[i]] <= k) return true;
            seen[nums[i]] = i;
        }

        return false;
    }
};

File: 219. Contains Duplicate II.java

import java.util.HashMap;
import java.util.Map;

class Solution {
    public boolean containsNearbyDuplicate(int[] nums, int k) {
        Map<Integer, Integer> seen = new HashMap<>();
        int n = nums.length;
        
        for(int i = 0; i < n; ++i){
            if(seen.containsKey(nums[i]) && i - seen.get(nums[i]) <= k) return true;
            seen.put(nums[i], i);
        }

        return false;
    }
}